src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -685,11 +697,8 @@ private async Task DismissUIAndRollbackEditsAndEndRenameSessionAsync(
         _keepAliveSession.Dispose();
 
         // Perform the actual commit step if we've been asked to.
-        if (finalCommitAction != null)
-        {
-            // ConfigureAwait(true) so we come back to the UI thread to finish work.
-            await finalCommitAction().ConfigureAwait(true);
-        }
+        if (finalSolution != null)
+            await ApplyChangesAsync(finalSolution).ConfigureAwait(true);
```

Comment:
this function is now only async for the 'commit' codepath, for teh 'cancel' codepath, it is entirely synchronous.